engines/scalac/src/main/scala/scala/macros/internal/engines/scalac/trees/Abstracts.scala

```diff
@@ -193,126 +217,154 @@ trait Abstracts extends scala.macros.trees.Abstracts with Positions { self: Univ
           if (targs.nonEmpty) method = g.TypeApply(method, targs).setPos(op.pos)
           g.Apply(method, args)
         } else {
+          val termName = g.freshTermName()(g.currentFreshNameCreator)
```

Comment:
We should be fine without the synthetic/artifact part. If it becomes a problem, we'll address that.